Makefile: platforms joining doesn't work

Currently, trying to run locally `make release` fails with the following error for me:
```
bgp-node-selector-amd64: digest: sha256:bf058f9eef5ce126feb1d8d097f2d6597917d997481c770fd132772125e37212 size: 739
make[2]: Leaving directory '/home/invidian/repos/kinvolk/packet-ccm'
# path to credentials based on manifest-tool's requirements here https://github.com/estesp/manifest-tool#sample-usage
/home/invidian/go/bin/manifest-tool push from-args --platforms linux/arm64 linux/amd64, --template quay.io/invidian/packet-ccm:bgp-node-selector-ARCH --target quay.io/invidian/packet-ccm:bgp-node-selector
FATA[0000] You must specify all three arguments --platforms, --template and --target
make[1]: *** [Makefile:198: push-manifest] Error 1
make[1]: Leaving directory '/home/invidian/repos/kinvolk/packet-ccm'
make: *** [Makefile:267: release] Error 2
```

--- a/Makefile
+++ b/Makefile
@@ -11,6 +11,8 @@ VERSION := $(RELEASE_TAG)-$(VERSION)
 endif
 GO_FILES := $(shell find . -type f -not -path './vendor/*' -name '*.go')
 BUILD_TAG ?= latest
+TAGGED_IMAGE ?= $(BUILD_IMAGE):$(BUILD_TAG)
+TAGGED_ARCH_IMAGE ?= $(TAGGED_IMAGE)-$(ARCH)
 LDFLAGS ?= -ldflags '-extldflags "-static" -X "$(PACKAGE_NAME)/version.VERSION=$(VERSION)"'
 
 # which arches can we support
@@ -161,8 +163,8 @@ sub-image-%:
 	@$(MAKE) ARCH=$* image
 
 image: ## make the image for a single ARCH
-	docker buildx build --load -t $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
-	echo "Done. image is at $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH)"
+	docker buildx build --load -t $(TAGGED_ARCH_IMAGE) -f Dockerfile --platform $(OS)/$(ARCH) .
+	echo "Done. image is at $(TAGGED_ARCH_IMAGE)"
 
 # Targets used when cross building.
 .PHONY: register

--- a/Makefile
+++ b/Makefile
@@ -166,6 +166,26 @@ image: ## make the image for a single ARCH
 	docker buildx build --load -t $(TAGGED_ARCH_IMAGE) -f Dockerfile --platform $(OS)/$(ARCH) .
 	echo "Done. image is at $(TAGGED_ARCH_IMAGE)"
 
+push-all: $(addprefix push-arch-, $(ARCHES)) ## Push all built images.
+push-arch-%:
+	@$(MAKE) ARCH=$* push
+
+push: image ## Push image to registry for a single ARCH.
+	docker push $(TAGGED_ARCH_IMAGE)
+
+manifest-push: manifest-all ## Make single image manifest for all supported ARCH and push it to registry.
+	docker manifest push $(TAGGED_IMAGE)
+
+manifest-all: manifest-create $(addprefix manifest-annotate-arch-, $(ARCHES)) ## Annotate docker manifest with all supported ARCH.
+manifest-annotate-arch-%:
+	@$(MAKE) ARCH=$* manifest-annotate
+
+manifest-annotate:
+	docker manifest annotate $(TAGGED_IMAGE) $(TAGGED_ARCH_IMAGE) --arch=$(ARCH) --os=$(OS)
+
+manifest-create: push-all ## Creates Docker manifest for all created images.
+	docker manifest create $(TAGGED_IMAGE) $(addprefix --amend $(TAGGED_IMAGE)-, $(ARCHES))
+
 # Targets used when cross building.
 .PHONY: register
 # Enable binfmt adding support for miscellaneous binary formats.

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ ifneq (,$(RELEASE_TAG))
 VERSION := $(RELEASE_TAG)-$(VERSION)
 endif
 GO_FILES := $(shell find . -type f -not -path './vendor/*' -name '*.go')
-FROMTAG ?= latest
+BUILD_TAG ?= latest
 LDFLAGS ?= -ldflags '-extldflags "-static" -X "$(PACKAGE_NAME)/version.VERSION=$(VERSION)"'
 
 # which arches can we support
@@ -161,8 +161,8 @@ sub-image-%:
 	@$(MAKE) ARCH=$* image
 
 image: ## make the image for a single ARCH
-	docker buildx build --load -t $(BUILD_IMAGE):latest-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
-	echo "Done. image is at $(BUILD_IMAGE):latest-$(ARCH)"
+	docker buildx build --load -t $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
+	echo "Done. image is at $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH)"
 
 # Targets used when cross building.
 .PHONY: register

